src/lib/__tests__/chartStorage.test.js

```diff
@@ -0,0 +1,152 @@
+const {
+    saveChart,
+    loadAllSavedCharts,
+    loadSavedChart,
+    updateCurrentChartData,
+    loadCurrentChartData
+} = require('../chartStorage');
+
+
+
+
+
+// Mock localStorage
+const localStorageMock = (() => {
+    let store = {};
+    return {
+        getItem: key => store[key] || null,
+        setItem: (key, value) => store[key] = value.toString(),
+        clear: () => store = {}
+    };
+})();
+global.localStorage = localStorageMock;
+
+// Mock window object
+global.window = {
+    localStorage: localStorageMock
+};
+
+
+
+describe('saveChart function', () => {
+    beforeEach(() => {
+        // Clear localStorage before each test
+        localStorage.clear();
+    });
+
+    test('adds a new chart to localStorage', () => {
+        // Define a mock chart
+        const chart = { data: [1, 2, 3], title: 'Test Chart' };
+
+        // Call saveChart function
+        saveChart(chart);
+
+        // Retrieve saved charts from localStorage
+        const savedCharts = loadAllSavedCharts();
+
+        // Expect the chart to be saved correctly
+        expect(savedCharts).toHaveLength(1);
+        expect(savedCharts[0]).toEqual(chart);
+    });
+
+ 
+});
+
+
+describe('loadCurrentChartData function', () => {
+    test('returns current chart data from localStorage', () => {
+        // Define mock current chart data
+        const currentChartData = { data: [4, 5, 6], title: 'New Chart' };
+
+        // Store mock current chart data in localStorage
+        window.localStorage.setItem('currentChartData', JSON.stringify(currentChartData));
+
+        // Call loadCurrentChartData function
+        const loadedData = loadCurrentChartData();
+
+        // Expect the current chart data to be loaded correctly
+        expect(loadedData).toEqual(currentChartData);
+    });
+
+});
+
+
+test('updates the current chart data in localStorage without changing the length', () => {
```

Comment:
One recommendation I would have or unless you've tested this individually is trying to have some tests use more data. For example, instead of data having three values have some of the data to contain more values.